dot/sync/chain_sync_test.go

```diff
@@ -0,0 +1,1051 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	syncmocks "github.com/ChainSafe/gossamer/dot/sync/mocks"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/common/variadic"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/golang/mock/gomock"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_chainSyncState_String(t *testing.T) {
+	tests := []struct {
+		name string
+		s    chainSyncState
+		want string
+	}{
+		{
+			name: "case bootstrap",
+			s:    bootstrap,
+			want: "bootstrap",
+		},
+		{
+			name: "case tip",
+			s:    tip,
+			want: "tip",
+		},
+		{
+			name: "case unknown",
+			s:    3,
+			want: "unknown",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.s.String(); got != tt.want {
+				t.Errorf("String() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+const (
+	defaultMinPeers     = 1
+	defaultMaxPeers     = 5
+	testTimeout         = time.Second * 5
+	defaultSlotDuration = time.Second * 6
+)
+
+func newTestChainSync(t *testing.T) (*chainSync, *blockQueue) {
+	header, err := types.NewHeader(common.NewHash([]byte{0}), trie.EmptyHash, trie.EmptyHash, 0,
+		types.NewDigest())
+	require.NoError(t, err)
+
+	bs := new(syncmocks.BlockState)
+	bs.On("BestBlockHeader").Return(header, nil)
+	bs.On("GetFinalisedNotifierChannel").Return(make(chan *types.FinalisationInfo, 128), nil)
+	bs.On("HasHeader", mock.AnythingOfType("common.Hash")).Return(true, nil)
+
+	net := new(syncmocks.Network)
+	net.On("DoBlockRequest", mock.AnythingOfType("peer.ID"), mock.AnythingOfType("*network.BlockRequestMessage")).
+		Return(nil, nil)
+	net.On("ReportPeer", mock.AnythingOfType("peerset.ReputationChange"), mock.AnythingOfType("peer.ID"))
+
+	readyBlocks := newBlockQueue(maxResponseSize)
```

Comment:
this should be taken outside of the `newTestChainSync`, it's rather independent I think.

Plus we often don't use it, i.e. `cs, _ := newTestChainSync(t)`

Reply:
I disagree.

Reply:
Why?

There are 16 instances where `readyBlocks` is unused (returned and taken as `_`), and only 2 where it's used.

Reply:
Refactored netTestChainSync and moved to test_helpers.go